homeassistant/components/screenlogic/entity.py

```diff
@@ -0,0 +1,143 @@
+"""Base ScreenLogicEntity definitions."""
+import logging
+
+# from screenlogicpy import ScreenLogicError, ScreenLogicGateway
+from screenlogicpy.const import DATA as SL_DATA, EQUIPMENT, ON_OFF
+
+from homeassistant.core import callback
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import ScreenlogicDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ScreenlogicEntity(CoordinatorEntity[ScreenlogicDataUpdateCoordinator]):
+    """Base class for all ScreenLogic entities."""
+
+    def __init__(self, coordinator, data_key, enabled=True):
+        """Initialize of the entity."""
+        super().__init__(coordinator)
+        self._data_key = data_key
+        self._enabled_default = enabled
+
+    @property
+    def entity_registry_enabled_default(self):
+        """Entity enabled by default."""
+        return self._enabled_default
+
+    @property
+    def mac(self):
+        """Mac address."""
+        return self.coordinator.config_entry.unique_id
+
+    @property
+    def unique_id(self):
+        """Entity Unique ID."""
+        return f"{self.mac}_{self._data_key}"
+
+    @property
+    def config_data(self):
+        """Shortcut for config data."""
+        return self.coordinator.gateway.get_data()[SL_DATA.KEY_CONFIG]
+
+    @property
+    def gateway(self):
+        """Return the gateway."""
+        return self.coordinator.gateway
+
+    @property
+    def gateway_name(self):
+        """Return the configured name of the gateway."""
+        return self.gateway.name
+
+    @property
+    def device_info(self) -> DeviceInfo:
```

Comment:
Since you are refactoring you could use self._attr_device_info = DeviceInfo(...

In __init__ instead

Reply:
Was also thinking about building the DeviceInfo in the coordinator and just referencing it in the entity. Would save a bunch of properties in the base entity class.

Reply:
That works as well